server/matchmaker/matchmaker_queue.py

```diff
@@ -152,14 +154,27 @@ async def find_matches(self) -> None:
         if self.num_players < 2 * self.team_size:
             return
 
-        searches = self.find_teams()
+        searches = list(self._queue.keys())
 
         # Call self.match on all matches and filter out the ones that were cancelled
         loop = asyncio.get_running_loop()
-        matches = list(filter(
-            lambda m: self.match(m[0], m[1]),
-            await loop.run_in_executor(None, make_matches, searches)
-        ))
+        proposed_matches, unmatched_searches = await loop.run_in_executor(
+            None,
+            self.matchmaker.find,
+            searches,
+            self.team_size,
+        )
+
+        # filter out matches that were cancelled
+        matches: List[Match] = []
+        for match in proposed_matches:
```

Comment:
You could unpack here so you don't need to index into `match` later
```suggestion
        for s1, s2 in proposed_matches:
```